pkg/controller/build/imagebuilder/jobimagebuilder.go

```diff
@@ -0,0 +1,146 @@
+package imagebuilder
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	mcfgv1alpha1 "github.com/openshift/api/machineconfiguration/v1alpha1"
+	mcfgclientset "github.com/openshift/client-go/machineconfiguration/clientset/versioned"
+	"github.com/openshift/machine-config-operator/pkg/controller/build/buildrequest"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+// I AM NOT COMPLETELY IMPLEMENTED YET!!!
+type jobImageBuilder struct {
+	*baseImageBuilder
+	cleaner Cleaner
+}
+
+func newJobImageBuilder(kubeclient clientset.Interface, mcfgclient mcfgclientset.Interface, mosb *mcfgv1alpha1.MachineOSBuild, mosc *mcfgv1alpha1.MachineOSConfig, builder buildrequest.Builder) *jobImageBuilder {
```

Comment:
@umohnani8 I didn't want to focus on getting this implementation done since I know you've been doing a lot of thinking about it. I just wanted to bring the existence of this to your attention :smile: